package-deps-hash/src/getPackageDeps.ts

```diff
@@ -11,63 +11,61 @@ export function getPackageDeps(packagePath: string = process.cwd(), excludedPath
   }
 
   return new Promise((complete) => {
-    child_process.exec(
+    const stdout: string = child_process.execSync(
       `git ls-tree HEAD -r && echo ${PROCESS_OUTPUT_DELIMITER} && git status -s -u .`,
-      { cwd: packagePath },
-      (error: Error, stdout: string) => {
-        const changes: IPackageDeps = {
-          files: {}
-        };
-        const processOutputBlocks: string[] = stdout.split(PROCESS_OUTPUT_DELIMITER);
+      { cwd: packagePath }).toString();
 
-        // Note: The output of git ls-tree uses \n newlines regardless of OS.
-        processOutputBlocks[0].split('\n').forEach(line => {
-          if (line) {
-            const parts: string[] = line.substr(line.indexOf('blob ') + 5).split('\t');
-            if (!excludedHashes[parts[1]]) {
-              changes.files[parts[1]] = parts[0];
-            }
-          }
-        });
+    const changes: IPackageDeps = {
+      files: {}
+    };
+    const processOutputBlocks: string[] = stdout.split(PROCESS_OUTPUT_DELIMITER);
```

Comment:
So it looks like processOutputBlocks just splits the output (with the delimiter) into 2 arrays. So it should only have 1 element if the delimiter was not written, which would occur if the `git ls-tree` command returns a non-zero exit code. However, the way split works, is that the entire string goes into `processOutputBlocks[0]` if the delimiter is not found.

Either way, we check the value of `processOutputBlocks[1]` before doing anything.

I can add some comments about that.